From 629d8a0ecc2b364bc3bd9bb7284b8fed65ff7459 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 May 2020 11:54:55 +0300
Subject: [PATCH] [SW-9973] gaudi: override flash protection mask

Use Gaudi specific protection mask which is 512KB from start of flash.

Change-Id: Ibbae06370e66aaa4cea70688945849135f1a9e61
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index d35f70d..c818647 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -11,6 +11,7 @@
 #include <psoc_spi_regs.h>
 
 #include "preboot.h"
+#include "common.h"
 #include "spi.h"
 #include "uart.h"
 
diff --git a/gaudi/common.h b/gaudi/common.h
index 89b69fe..4869859 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -39,4 +39,15 @@
 #define FUSE_WORDS_PER_BANK		0x10
 #define I2C_SLAVE_ADDRESS		0x4f
 
+#ifndef __ASSEMBLY__
+#include "spi.h"
+
+/* Let this file be included in any order and override the PROT_MASK */
+#ifdef PROT_MASK
+#undef PROT_MASK
+#endif
+/* Only BP2 == 0x4 - means 8 blocks - up to 512KB */
+#define PROT_MASK			READ_STATUS_BP2
+
+#endif /* __ASSEMBLY__ */
 #endif /* COMMON_GAUDI_H_ */
